packages/eas-cli/src/utils/code-signing.ts

```diff
@@ -0,0 +1,169 @@
+import {
+  convertCertificatePEMToCertificate,
+  convertPrivateKeyPEMToPrivateKey,
+  signStringRSASHA256AndVerify,
+  validateSelfSignedCertificate,
+} from '@expo/code-signing-certificates';
+import Dicer from 'dicer';
+import isDeepEqual from 'fast-deep-equal';
+import { promises as fs } from 'fs';
+import { Response } from 'got';
+import { pki as PKI } from 'node-forge';
+import nullthrows from 'nullthrows';
+import { Stream } from 'stream';
+import { parseItem } from 'structured-headers';
+
+import { PartialManifest, PartialManifestAsset } from '../graphql/generated';
+
+export async function getKeyAndCertificateFromPathsAsync({
+  codeSigningCertificatePath,
+  privateKeyPath,
+}: {
+  codeSigningCertificatePath: string;
+  privateKeyPath: string;
+}): Promise<{ privateKey: PKI.rsa.PrivateKey; certificate: PKI.Certificate }> {
+  const [codeSigningCertificatePEM, privateKeyPEM] = await Promise.all([
+    fs.readFile(codeSigningCertificatePath, 'utf8'),
+    fs.readFile(privateKeyPath, 'utf8'),
+  ]);
+
+  const privateKey = convertPrivateKeyPEMToPrivateKey(privateKeyPEM);
+  const certificate = convertCertificatePEMToCertificate(codeSigningCertificatePEM);
+  validateSelfSignedCertificate(certificate, {
+    publicKey: certificate.publicKey as PKI.rsa.PublicKey,
+    privateKey,
+  });
+
+  return {
+    privateKey,
+    certificate,
+  };
+}
+
+export type MultipartPart = { headers: Map<string, string>; body: string };
+
+export async function parseMultipartMixedResponseAsync(res: Response): Promise<MultipartPart[]> {
+  const contentType = res.headers['content-type'];
+  if (!contentType) {
+    throw new Error('Missing content-type in multipart response');
```

Comment:
Since these errors might show up to developers, making them a little more human-readable will make error messages sent our way easier to understand:
```suggestion
    throw new Error('The multipart response is missing the content-type header');
```